### PR TITLE
add options back to diagnostics

### PR DESCRIPTION
**Pull request by heejaechang**
looks like my previous refactoring - [link](https://github.com/dotnet/roslyn/commit/b4941197495edfef28698246ff5303c1cfa99ab3#diff-34b1790434d03262ef0e31a49e147b33R36) - removed options

this put options back.

I am working on enabling d15 unit tests back on in another PR which will include unit test to prevent this from happening again.

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani @panopticoncentral @CyrusNajmabadi @KevinH-MS can you take a look?

File: src/Workspaces/Remote/ServiceHub/Services/ServiceHubServiceBase.cs [hunk 1]
```diff
@@ -60,10 +61,10 @@ protected RoslynServices RoslynServices
             }
         }
 
-        protected Task<Solution> GetSolutionAsync()
+        protected Task<Solution> GetSolutionAsync(OptionSet options = null)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Services/ServiceHubServiceBase.cs]: Can we make things non-optional?  It makes it super simple for someone to accidently do the wrong thing.  We shoudl probably document all the spots that pass null and explain why that's the right thing to do.

Comment (heejaechang) [inline src/Workspaces/Remote/ServiceHub/Services/ServiceHubServiceBase.cs]: right now, option is a bit weird unfortunately until option becomes part of snapshot. how about adding new API GetSolutionWithSpecificOptionsAsync

**Comment by heejaechang** [comment]
retest this please

**Comment by heejaechang** [comment]
retest due to Jenkins outrage.

**Comment by heejaechang** [comment]
ping?

**Comment by davkean** [comment]
So, why isn't this in RC2 build?

**Pull request merged**